File: 백준/Bronze/2920. 음계/음계.py
def ascending():
    global result
    if arr == [1, 2, 3, 4, 5, 6, 7, 8]:
        result = 'ascending'

def descending():
    global result
    if arr == [8, 7, 6, 5, 4, 3, 2, 1]:
        result = 'descending'

arr = list(map(int, input().split()))
result = ''
ascending()

descending()

if result == '':
    print('mixed')
else:
    print(result)

File: 백준/Bronze/2920. 음계/음계.cc
#include <iostream>
using namespace std;
 
int main(){
    int n;
    cin >> n;
 
    if(n == 1){
        for(int i = 2; i < 9; i++){
            cin >> n;
            if(i != n){
                cout << "mixed";
                break;
            }
            if(n == 8) 
                cout << "ascending";
        }
    } else if(n == 8){
        for(int i = 7; i > 0; i--){
            cin >> n;
            if(i != n){
                cout << "mixed";
                break;
            }
            if(n == 1) 
                cout << "descending";
        }
    } else cout << "mixed";
    return 0;
}